fdbclient/ManagementAPI.actor.cpp

```diff
@@ -1341,13 +1341,17 @@ ACTOR Future<vector<AddressExclusion>> getExcludedServers( Database cx ) {
 
 ACTOR Future<Void> checkDataDistributionStatus(Database cx, bool printWarningOnly) {
 	state Transaction tr(cx);
+	state Future<Void> timeoutDelay = printWarningOnly ? delay(2.0) : Never();
 	loop {
 		try {
 			tr.setOption(FDBTransactionOptions::LOCK_AWARE);
 			state Future<Optional<Value>> overallSwitchF = tr.get(dataDistributionModeKey);
 			state Future<Optional<Value>> healthyZoneValueF = tr.get(healthyZoneKey);
 			state Future<Optional<Value>> rebalanceDDIgnoreValueF = tr.get(rebalanceDDIgnoreKey);
-			wait(success(overallSwitchF) && success(healthyZoneValueF) && success(rebalanceDDIgnoreValueF));
+			wait(timeoutDelay || (success(overallSwitchF) && success(healthyZoneValueF) && success(rebalanceDDIgnoreValueF)));
+			if(timeoutDelay.isReady()) {
+				return Void();
```

Comment:
This will just silently not check the status if it times out. I think we should print out a warning that we were unable to fetch it.

We could arguably also increase the timeout, as the status check that this runs concurrently with will try for longer.

Reply:
This is a bug fix.

The timeout is only used when printWarningOnly is turned on. PrintWarningOnly is only used when fdbcli is just started. The problem was that if the database was down when you started fdbcli you would never be taken to a command prompt. This was particularly problematic in the case of configuring a brand new database, because you would never be given the option to type "configure new single memory".

I did not think we needed to print anything, because this command will be run concurrently with regular status, which should tell the user that fdbcli is down.

Reply:
The problem is that regular status could return a result after 2 seconds but which indicates things are ok. The absence of any message from DD status could mislead a user to believe things about data distribution that aren't true, so I think my preference would be that we indicate that we were unable to get the status.

Reply:
As an example, if GRV latencies are 4 seconds, I think status will work but the data distribution check will time out.

Reply:
The purpose of this warning is to remind a user that they have disabled data distribution. Given that we do not officially support disabling data distribution (it is a hidden command), it just feels a little weird to warn about the check failing to complete. 

Maybe a better location for the warning is in regular status, rather than at the beginning of fdbcli?

Reply:
I agree with Evan on removing the status check on starting up the cli. It was added without much thought.

Reply:
Yeah that makes sense to me. I like the idea of making it part of normal status (or even just contingent upon successful status) if we wanted to keep it.